frontend/src/features/env/mutations.ts

```diff
@@ -1,28 +1,52 @@
 // TODO #4279: Remove after React rollout is complete
 import { useMutation } from 'react-query'
-import { useNavigate } from 'react-router-dom'
+import { useParams } from 'react-router-dom'
+
+import { PublicFormViewDto, switchEnvFeedbackFormBodyDto } from '~shared/types'
 
 import {
   adminChooseEnvironment,
   publicChooseEnvironment,
 } from '~services/EnvService'
 
-export const useEnvMutations = () => {
-  const navigate = useNavigate()
+import { submitSwitchEnvFormFeedback } from './EnvService'
+
+export const useEnvMutations = (
+  feedbackForm: PublicFormViewDto | undefined,
+) => {
+  const { formId } = useParams()
 
   const publicSwitchEnvMutation = useMutation(() => publicChooseEnvironment(), {
     onSuccess: () => window.location.reload(),
   })
 
   const adminSwitchEnvMutation = useMutation(() => adminChooseEnvironment(), {
     onSuccess: () => {
-      navigate('/#!/forms')
-      window.location.reload()
+      // If on form builder page
+      if (window.location.href.indexOf('admin/form/') > -1) {
```

Comment:
I'm always worried of `href.indexOf()` checks, because they can capture stuff in query string or hash, which should have been inconsequential to the actual test. Better to be very specific on the source string (check the properties [`window.location`](https://developer.mozilla.org/en-US/docs/Web/API/Location) offers), and by using a regexp, preferably with an anchor if possible.